testsuite/tests/singlecluster/authorino/metrics/conftest.py

```diff
@@ -40,33 +38,17 @@ def prometheus(cluster):
 
 
 @pytest.fixture(scope="module")
-def gateway(request, authorino, cluster, blame, label, testconfig) -> Envoy:
-    """Deploys Envoy that wires up the Backend behind the reverse-proxy and Authorino instance"""
-    gw = Envoy(
-        cluster,
-        blame("gw"),
-        authorino,
-        testconfig["service_protection"]["envoy"]["image"],
-        labels={"app": label},
-    )
-    request.addfinalizer(gw.delete)
-    gw.commit()
-    gw.wait_for_ready()
-    return gw
-
-
-@pytest.fixture(scope="module")
-def authorino(authorino, module_label):
-    """Label Authorino controller-metrics service for the proper discovery"""
-    authorino.metrics_service.label({"app": module_label})
-    return authorino
-
-
-@pytest.fixture(scope="module")
-def service_monitor(cluster, prometheus, blame, module_label):  # pylint: disable=unused-argument
+def service_monitor(cluster, prometheus, blame, module_label, authorino, kuadrant):  # pylint: disable=unused-argument
     """Create ServiceMonitor object to follow Authorino /metrics and /server-metrics endpoints"""
+    if kuadrant:
+        label = {"app": "service_monitor_label"}
```

Comment:
sidenote: if we are also in charge of choosing the label (I suspect we are since this is not a default one), why not use module label as well?

Reply:
Two parallel runs of the same test would be disruptive (two service monitors would target two different labels of the same service). Currently, that is not a problem but if we add more tests (which is not probable rn), different label would make this disruptive.
What do you think, should I use module label?

Reply:
And now they would not be?